### ISSUE TITLE
Null ApplicationId in AWS::Serverless::Application fails with Internal Server Error

### ISSUE THREAD

**Issue by ylynn** [awslabs/serverless-application-model#1053]
**Description:**

Null ApplicationId in AWS::Serverless::Application fails with unhandled exception

Template:
```
Resources:
  Application:
    Type: AWS::Serverless::Application
    Properties:
      Location:
        ApplicationId: null
        SemanticVersion: 1.0.0
```
errorMessage=Internal Server Error: Parameter validation failed:
Invalid type for parameter ApplicationId, value: None, type: <type 'NoneType'>, valid types: <type 'basestring'>

**Comment by jlhood** [comment]
This should be a really simple bugfix if anyone from the community wants to contribute. The validation check should be added after [this check](https://github.com/awslabs/serverless-application-model/blob/develop/samtranslator/plugins/application/serverless_app_plugin.py#L84).

**Comment by keetonian** [comment]
Will be released in 1.14.0

**Comment by ShreyaGangishetty** [comment]
Closing this issue as this is released in v1.14.0
